ams/server/src/main/java/com/netease/arctic/server/manager/MetricsManager.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *  *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.server.manager;
+
+import com.netease.arctic.ams.api.metrics.MetricsContent;
+import com.netease.arctic.ams.api.metrics.MetricsEmitter;
+import com.netease.arctic.server.exception.LoadingPluginException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.Map;
+
+public class MetricsManager extends ActivePluginManager<MetricsEmitter> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MetricsManager.class);
+
+  private final String configPath;
+
+  public MetricsManager(String configPath) {
+    this.configPath = configPath;
+  }
+
+  @Override
+  protected Map<String, String> loadProperties(String pluginName) {
+    try {
+      FileInputStream inputStream = new FileInputStream(configPath +
+          "/" + pluginName + ".yaml");
+      return new Yaml().load(inputStream);
+    } catch (FileNotFoundException e) {
+      throw new LoadingPluginException("Cannot load plugin " + pluginName, e);
+    }
+  }
+
+  public void emit(MetricsContent<?> metrics) {
```

Comment:
Is it appropriate for the MetricsManager to be responsible for both plugin loading and how plugins are used?

Reply:
This is the original interface in master. I haven't made any changes.

For more use cases. I thought calling by manager instead of plugins themselves could provide more delegate functions like ClassLoader context switch and hide picking plugin logic inside managers

And list() and iterator() are more usable for CLI tools

Reply:
> For more use cases. I thought calling by manager instead of plugins themselves could provide more delegate functions like ClassLoader context switch and hide picking plugin logic inside managers

Sounds good. BTW, The logic of plugins can be complex, and it can be difficult to restrict all calls within the Manager.

Reply:
Yes, that depends